### PR TITLE
refactor: update lints to embark lints 0.3

### PR DESCRIPTION
**Pull request by DeanBDean**
### Checklist

* [x] I have read the [Contributor Guide](../../CONTRIBUTING.md)
* [x] I have read and agree to the [Code of Conduct](../../CODE_OF_CONDUCT.md)
* [x] I have added a description of my changes and why I'd like them included in the section below

### Description of Changes

Update the linting rules to 0.3.

### Related Issues

Resolves this issue https://github.com/EmbarkStudios/tame-oidc/issues/8

### CODE REVIEW

File: src/deserialize_uri.rs [hunk 1]
```diff
@@ -11,6 +11,9 @@ impl<'de> de::Visitor<'de> for UriVisitor {
     }
 
     fn visit_str<E: de::Error>(self, val: &str) -> Result<Self::Value, E> {
+        #[allow(clippy::map_err_ignore)]
+        // Allowing because the intention is to ignore the previous error, and it would
+        // be impossible to implement one foreign trait on another
         val.parse()
             .map_err(|_| de::Error::invalid_value(de::Unexpected::Str(val), &self))
```
Comment (lpil) [inline src/deserialize_uri.rs:18]: Does clippy pass if we do this?

```suggestion
        val.parse()
            .map_err(|_err| de::Error::invalid_value(de::Unexpected::Str(val), &self))
```

**Review by lpil — COMMENTED** [review]
Thank you. I let a question below about the error ignore lint. If that works it would be good to use this approach in both places

File: src/deserialize_uri.rs (hunk 1 above)
Comment (DeanBDean) [reply src/deserialize_uri.rs:18]: Yes it does, changing to this

**Review by vojd — APPROVED** [review]
Looks fine! Took it for a test run and it worked as intended.
Thanks for contributing.

**Review by lpil — APPROVED** [review]
Wonderful, thank you very much! ✨

**Pull request merged**